sdk/src/account_api.rs

```diff
@@ -0,0 +1,295 @@
+//! Defines account apis for use in Solana programs.
+
+use crate::account::KeyedAccount;
+use crate::credit_only_account::KeyedCreditOnlyAccount;
+use crate::instruction::InstructionError;
+use crate::pubkey::Pubkey;
+use std::fmt::Debug;
+use std::sync::atomic::Ordering;
+
+#[derive(Debug)]
+pub enum AccountWrapper<'a> {
```

Comment:
Overall I like the direction this is going in!  As I read this patch I find myself wishing that:
1. KeyedAccount was called KeyedCreditDebitAccount (and Account was called CreditDebitAccount)
2. AccountWrapper didn’t exist
3. The AccountApi trait then renamed to KeyedAccount